### PR TITLE
Add non-verbose DAS Req/Resp logger 

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Implement DAS req/resp centralized logger which:
- Logs request/responses with a single log line
- Tries to be as descriptive as possible from one side and not flood logs with overwhelming amount of data

### CODE REVIEW

**Comment by Nashatyrev** [comment]
Sample log entries: 
```log
2024-10-17 17:56:03.490+04:00 | p2p-async-6 | DEBUG | DasReqRespLogger | ReqResp outbound data_column_sidecars_by_root, columns: 0/563 in 5002 ms with ERROR, peer 0x988ba340...1cdc2e32: request: 0x1298709154315a1d9c1973358381abf3d03d1d96323b5708e53836acffa02b10: 24, 0x21033f95b2f51b38e0b4400fb9333febb362e73843e59764d59a12e6150037bc: 13, 0x2829b588a8eeaf709d8add44bb165cbd052a65204f6c467f5cf2fd700c823cf9: 28, 0x2dbd27f5c3c543879b9b4ef09d88849a9fef6dfdcb6fca6ea116fe44ceb875d5: 18, 0x3151fc614e4b6ee9e1b021323132df6dd92e129ec56d4b301235f50b972ae1c4: 22, 0x36937a6edf11423ed31f8cfb9f0b83e52a26f28334f955ddf861f8404c987341: 27, 0x455550085ee83cb1a265b602e09c8db68c3a5c87fd1b3be902b8cf306e3d7802: 13, 0x5987bcb5369290af9f416a3724c3651b76ab29d78d6ad79d064c1d7e4f5aae9d: 35, 0x5daaff7d44d84a27cadc4d6a4a24c066bc36953cf7cdca4e525774764de47829: 20, 0x6cf02f896cbca1640fd673269621b116c9f590a205e32b097f49a1f027b56775: 17, 0x74c7ec5ca3b6814a2bdf5812c76c3af9e73aacfb77198d9028decad8923ce46a: 26, 0x859b8aeede2b47ef36b0bf6b9a9948b3badd91bb5272c7973f42e8a8a2c2cb71: 11, 0x8e241be6a268296c42c792b972c1b6702d32435681cb58ebddf73e0d66e46c02: 3, 0x9267062f9039a85c01d63cadea12840f66083042a17e293c1c8071b0e3c05beb: 26, 0x99c5f6a3b1a67c1113f1d40009d787436153fd7e1a2300f04d346b7a2112c5ad: 13, 0xa1529e870fc1226dc92c5b420b0dc4696f6859646838db922990268031f8027a: 22, 0xa4d7024160a09e7c81fa3f91efdfc540cca68000fc6f1d076a72aae10ea5bd3e: 1, 0xa7181a43fbc4d56c950b9f15ac4f4cbedfea76a7f249fc1a07642c93e4c0dcf3: 15, 0xaf24104fc9f60f20b1af3ad9b5b9cfde1ea4db282ce060f6517eb8b25bddc67b: 18, 0xb00bb963339ec3de0bea235c493aec1ca2581500bcd93121109e3e470aaf6897: 18, 0xb278e96fa764ebbabc3c8f87480ca8ec1ab10aea4154de35f87c174a111eeeac: 1, 0xb61872829b124631d18276789d75398ba8ba8fb8880c936ff379fdc6fe3ad414: 17, 0xc248d4618dd9117942605a87de9ceda2fdb42ebba9637222329fddcefbd04734: 13, 0xc3364009342c2b3b3f44239c84c6bb99a523f4ec016c6c900a6f9bc1cedb6353: 28, 0xcdb55f39d286317788bbe55eff861b724ac0370bf2acbd2e812c2a82c1540f02: 19, 0xdb258ece35210cfcad81a849cf8903c9af87a943aef80ffe51d7f7e840fbd724: 25, 0xdb27f010641fbe039d00688bd2a21c80132bf37fd0dd866619f2ccde7d7a87f3: 22, 0xe385a885230cdc86ee56cfd361124142d9e41e604da2da6f7aca94dc4443a817: 19, 0xe8db5cdb7a91da57315e28187c2ceff761ed38b16f36d7e2d49494ac05b11bc3: 11, 0xf130d0016201de7dd3bdadefba5f05c36ee1c7b95762f992bdaa9caa9554794f: 13, 0xf4b9c370a43ccec0bdd990977cb2b82fd0b204c11fb4762b8fd66d183d11fe39: 19, 0xfa336b8938ac2b702195842476dfb7146347c3df4c32e4791cc46d030bb65612: 6, response: error: java.util.concurrent.CompletionException: tech.pegasys.teku.networking.p2p.rpc.StreamTimeoutException: Timed out waiting to initialize stream for protocol(s): /eth2/beacon_chain/req/data_column_sidecars_by_root/1/ssz_snappy
2024-10-17 17:56:04.491+04:00 | beaconchain-async-6 | DEBUG | DasReqRespLogger | ReqResp inbound data_column_sidecars_by_root, columns: 0/13 in 1 ms, peer 16Uiu2HAmAMLk69zmpWrN4ecDGcCekffBqeFVNwGPU8gZ2xuZAcZU (nodeId = 0x48a3d027...402e6e35): request: 13 columns: 0xa09c50a78fcf287c1ff7695d92cd10992e42b6284caffe57ed1528b32f9d62df colIdxs: (len: 13) [14,28,37,47,55,64,72,79,86,96,105,111,120], response: <empty>
2024-10-17 17:56:05.309+04:00 | p2p-async-3 | DEBUG | DasReqRespLogger | ReqResp outbound data_column_sidecars_by_root, columns: 1845/1845 in 11929 ms, peer 0x03fa68ed...f6bdd03a: request: #16038 (0x6cf02f..6775): 36, #16235 (0x8f1ef8..032b): 45, #16236 (0x6a3511..e75a): 41, #16237 (0x2292e3..16d7): 53, #16240 (0x3943e9..4dbf): 48, #16242 (0x7e2e75..ed7c): 48, #16243 (0xb278e9..eeac): 50, #16244 (0x8e241b..6c02): 44, #16245 (0xfbdf8c..2122): 44, #16248 (0x8c8596..1e9d): 43, #16249 (0xb61bdf..dd3e): 43, #16250 (0xe8db5c..1bc3): 39, #16251 (0x8ae8cd..2144): 44, #16253 (0x97707f..7977): 54, #16254 (0xf130d0..794f): 53, #16255 (0xfa336b..5612): 41, #16256 (0xa4d702..bd3e): 50, #16257 (0x3151fc..e1c4): 42, #16258 (0xc248d4..4734): 45, #16259 (0x455550..7802): 36, #16260 (0xa1529e..027a): 36, #16261 (0x99c5f6..c5ad): 46, #16262 (0x21033f..37bc): 54, #16263 (0xa7181a..dcf3): 42, #16264 (0x926706..5beb): 34, #16265 (0xc33640..6353): 41, #16266 (0xe385a8..a817): 40, #16267 (0x2829b5..3cf9): 48, #16269 (0xaf2410..c67b): 39, #16270 (0x74c7ec..e46a): 35, #16272 (0x2dbd27..75d5): 49, #16273 (0xb00bb9..6897): 49, #16274 (0x129870..2b10): 40, #16275 (0x5987bc..ae9d): 38, #16276 (0xcdb55f..0f02): 51, #16277 (0x5daaff..7829): 42, #16278 (0xdb258e..d724): 40, #16280 (0x36937a..7341): 43, #16281 (0xdb27f0..87f3): 48, #16282 (0xf4b9c3..fe39): 40, #16283 (0x859b8a..cb71): 48, #16284 (0xb61872..d414): 43, response: <all requested>
2024-10-17 17:56:34.034+04:00 | p2p-async-1 | DEBUG | DasReqRespLogger | ReqResp outbound data_column_sidecars_by_root, columns: 128/128 in 3469 ms, peer 0x48a3d027...402e6e35: request: 128 columns: #16309 (0x5abce1..3824) colIdxs: (len: 128) [all], response: <all requested>
```

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/DataColumnSidecarsByRangeMessageHandler.java [hunk 1]
```diff
@@ -85,36 +89,35 @@ public DataColumnSidecarsByRangeMessageHandler(
...
+                    message.getStartSlot(), message.getCount().intValue(), message.getColumns()));
+    LoggingResponseCallback<DataColumnSidecar> loggingCallback =
+        new LoggingResponseCallback<>(responseCallback, responseLogger);
 
     final int requestedCount = message.getMaximumResponseChunks();
 
     if (requestedCount > specConfigEip7594.getMaxRequestDataColumnSidecars()) {
       requestCounter.labels("count_too_big").inc();
-      callback.completeWithErrorResponse(
+      loggingCallback.completeWithErrorResponse(
```
Comment (zilm13) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/DataColumnSidecarsByRangeMessageHandler.java]: it's a bit confusing, maybe just callback or responseCallbackWithLogging

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/DataColumnSidecarsByRootMessageHandler.java [hunk 2]
```diff
@@ -102,23 +105,21 @@ public void onIncomingMessage(
...
-        "[nyota] DataColumnSidecarsByRootMessageHandler: REQUEST(#{}) {} data column sidecars from {}",
-        requestId,
-        message.size(),
-        peer.getId());
+    LoggingResponseCallback<DataColumnSidecar> loggingCallback =
+        new LoggingResponseCallback<>(responseCallback, responseLogger);
 
     final Optional<RequestApproval> dataColumnSidecarsRequestApproval =
-        peer.approveDataColumnSidecarsRequest(callback, message.size());
+        peer.approveDataColumnSidecarsRequest(loggingCallback, message.size());
```
Comment (zilm13) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/DataColumnSidecarsByRootMessageHandler.java]: same here with `loggingCallback`, I'd prefer to hide logging or put it in 2nd role in name

**Review by zilm13 — APPROVED** [review]
LGTM
But I worry how easy is to find logging level and logging string details starting from logger presensce in gossip handler

**Pull request merged**